magpie/adapter/magpieowssecurity.py

```diff
@@ -269,11 +269,18 @@ def update_request_cookies(self, request):
         """
         settings = get_settings(request)
         token_name = get_constant("MAGPIE_COOKIE_NAME", settings_container=settings)
+        network_mode = get_constant("MAGPIE_NETWORK_MODE", settings_container=settings,
+                                    settings_name="magpie.network_mode")
+        headers = dict(request.headers)
+        network_token_name = get_constant("MAGPIE_NETWORK_TOKEN_NAME", settings_container=settings)
+        if network_mode and "Authorization" not in headers and network_token_name in request.params:
+            headers["Authorization"] = "Bearer {}".format(request.params[network_token_name])
```

Comment:
> Why not use the Authorization: Bearer {token} header directly?

I think the idea here was to provide an option for cases when a user wants to access a resource and doesn't have the ability to set the request headers. For example, if they're using a tool that just asks for a URL.

Reply:
Ok, I can understand the logic behind it. If really necessary for tools with limited capabilities, that can be permitted. 
However, consider that this is less secure than headers. Path/Query are too often dumped in logs or directly exposed. 
If you can use headers, it is preferable because it will obfuscate the tokens slightly more.

Reply:
Yes, I completely agree. The short lifespan of these tokens is supposed to mitigate this risk a bit.